core/device_hooks/CMakeLists.txt

```diff
@@ -2,7 +2,7 @@ if(NOT GINKGO_BUILD_CUDA)
     add_library(ginkgo_cuda
         $<TARGET_OBJECTS:ginkgo_cuda_device>
         cuda_hooks.cpp)
-    target_link_libraries(ginkgo_cuda PUBLIC ginkgo_core_device)
+    target_link_libraries(ginkgo_cuda PRIVATE ginkgo_core_device)
```

Comment:
`core/device` does not really exist right ? So maybe we can change this library name to `ginkgo_device` or something like that ?